### ISSUE TITLE
`prefer-expect-assertions` doesn’t work with test context’s `expect`

### ISSUE THREAD

**Issue by davecardwell** [veritem/eslint-plugin-vitest#329]
Thank you for the new `vitest/prefer-expect-assertions` rule.

While testing I found that the rule is triggered when using the `test()` context argument’s version of `expect`:

```ts
import { describe, expect, test } from "vitest";

describe("prefer-expect-assertions", () => {
  // works as expected
  test("example", async () => {
    expect.assertions(1);
    await expect(Promise.resolve(null)).resolves.toBeNull();
  });

  // fails vitest/eslint-expect-assertions
  test("example", async ({ expect }) => {
    expect.assertions(1);
    await expect(Promise.resolve(null)).resolves.toBeNull();
  });
});

```
